### DISCUSSION TITLE
Question about identifying differentially accessible peaks and overrepresented motifs

### DISCUSSION THREAD

**Discussion by chunhuicai** [stuart-lab/signac#1533] [Q&A] [answered]
Hi,

We are currently utilizing Signac to identify overrepresented motifs by comparing different sample conditions in our study, specifically CNV, ASD, and CON. Our initial step involved using the FindMarkers function to identify differentially accessible peaks between 1) CNV and CON, and 2) ASD and CON. Subsequently, we searched for DNA motifs that are notably prevalent in these differentially accessible peaks.

Our analysis revealed a different number of differentially accessible peaks in different comparisons: 3,140 peaks when comparing CNV and CON, and 5,312 peaks for ASD and CON, with an overlap of 1,465 peaks between them. However, our attempt to identify overrepresented motifs in these peaks led to an unexpected result. We found 746 motifs, which correspond to the total number of motifs in the CORE collection, for both CNV vs. CON and ASD vs. CON comparisons. This also meant an overlap of 746 motifs. We even tried expanding our search beyond the CORE collection and different p value threshold for identifying peaks, but the outcome remained unchanged. I am attaching my code for your review and would greatly appreciate any suggestion you can provide in resolving this issue.

Thank you very much for your help!

Best regards,
Chunhui

## Get a list of motif position frequency matrices from the JASPAR database
pfm <- getMatrixSet(
  x = JASPAR2020,
  opts = list(collection = "CORE", tax_group = 'vertebrates', all_versions = FALSE)
)

## add motif information
multiomes.integrated <- AddMotifs(
  object = multiomes.integrated,
  genome = BSgenome.Hsapiens.UCSC.hg38,
  pfm = pfm
)

## Switch the identifiers to group
Idents(multiomes.integrated) <- multiomes.integrated$group

## Finding differentially accessible peaks when comparing ASD vs CON
da_peaks_ASDvsCON <- FindMarkers(
  object = multiomes.integrated,
  ident.1 = 'ASD',
  ident.2 = 'CON',
  only.pos = TRUE,
  test.use = 'LR',
  min.pct = 0.05,
  latent.vars = 'nCount_peaks'
)

## Finding differentially accessible peaks when comparing CNV vs CON
da_peaks_CNVvsCON <- FindMarkers(
  object = multiomes.integrated,
  ident.1 = 'CNV',
  ident.2 = 'CON',
  only.pos = TRUE,
  test.use = 'LR',
  min.pct = 0.05,
  latent.vars = 'nCount_peaks'
)

## get top differentially accessible peaks: ASDvsCON and CNVvsCON
top.da.peak_ASDvsCON <- rownames(da_peaks_ASDvsCON[da_peaks_ASDvsCON$p_val_adj<0.01, ])
top.da.peak_CNVvsCON <- rownames(da_peaks_CNVvsCON[da_peaks_CNVvsCON$p_val_adj<0.01, ])
length(top.da.peak_ASDvsCON) # 5,312
length(top.da.peak_CNVvsCON) # 3,140
length(intersect(top.da.peak_ASDvsCON, top.da.peak_CNVvsCON)) # 1,465

## Identify enriched motifs: ASD vs CON; CNV vs CON
DefaultAssay(multiomes.integrated) <- "peaks"
enriched.motifs_ASDvsCON <- FindMotifs(
  object = multiomes.integrated,
  features = top.da.peak_ASDvsCON
)
enriched.motifs_CNVvsCON <- FindMotifs(
  object = multiomes.integrated,
  features = top.da.peak_CNVvsCON
)
nrow(enriched.motifs_ASDvsCON) # 746
nrow(enriched.motifs_CNVvsCON) # 746
length(intersect(rownames(enriched.motifs_ASDvsCON), rownames(enriched.motifs_CNVvsCON))) # 746

**Comment by timoast** [comment, marked as answer]
`FindMotifs` will report the fold enrichment and p-value for **each** motif in the dataset, you need to look at those values to determine which motifs are differentially enriched between your peak sets. It's expected that the output will be the same dimension.

**Reply by chunhuicai** [reply]
Thank you for explaining the process of using FindMotifs to identify differentially enriched motifs in peak sets. I understand that the fold enrichment and p-value for each motif are key indicators. For my case majority of adjusted p values are less than 0.05. Could you please advise on what threshold of fold enrichment values should typically be considered significant in this context?

**Reply by timoast** [reply]
I don't any specific recommendations for a fold enrichment to use, you will need to decide what makes the most sense for your experiment
